Blogging Prompts: Add 'answered' indicator to my-home card

roo2 opened the pull request:
Part of pe7F0s-1iI-p2#comment-1700
In the mobile apps there is an indicator for when a user has answered a specific prompt, and we also get that data returned by both the v2 and v3 APIs.

This PR adds a simple indicator to the my home card if the user has answered a prompt.

<img width="557" alt="Screen Shot 2023-11-02 at 3 11 25 pm" src="https://github.com/Automattic/wp-calypso/assets/22446385/defc9734-97da-4fbd-b299-5dfd32044c1b">


I also just changed the vertical alignment of the prompt and prev/next buttons so that they don't jump around while cycling through prompts.

### Test instructions
Go to a site with `site_intent=write`
Answer a writing prompt.
Go to the `/home` page
You should see the "✅Answered" indicator on the prompt that you answered

roo2 committed:
add 'answered' indeicator to blogging prompts

Addison-Stavlo reviewed (commented):
I love the idea. 2 things.

1. Im not having much luck in testing this, after i publish an answer to a daily prompt and return back to home/refresh/wait/etc. the card is still not picking up the answered status.

2. I wonder if there is a better design/placement for this indication. cc @crisbusquets, @osullivanchris, @MindyPostoff if you have ideas.

crisbusquets commented:
Thanks for the ping, @Addison-Stavlo !

Perhaps we could change the text inside the button. If they've already answered it, the `Post Answer` button doesn't make sense.

If it's feasible, I'm leaning towards option C:
* the button shows that they've answered the daily writing prompt
* it's enabled and accessible
* if they click it, they will see their answer (I'm assuming there's a link there)

<img width="1235" alt="Screenshot 2023-11-02 at 16 34 20" src="https://github.com/Automattic/wp-calypso/assets/25607244/b70b6234-d82a-4b6d-b60b-b13d0539a0d2">

roo2 commented:
> Im not having much luck in testing this, after i publish an answer to a daily prompt and return back to home/refresh/wait/etc. the card is still not picking up the answered status.

Ahh I wonder if it's due to caching, I thought I would have issues because the responses are cached, but it seemed to work consistently for me. I'll look into if there's a way I can trigger a re fetch or change the caching

roo2 committed:
add 'answered' indeicator to blogging prompts

roo2 committed:
redesign answered indicator

roo2 commented:
I added a fix for the caching and updated to use the new design
<img width="565" alt="Screen Shot 2023-11-03 at 11 23 55 pm" src="https://github.com/Automattic/wp-calypso/assets/22446385/37deb8f8-4ecd-4f35-b70c-54d28700996d">

eoigal commented on client/components/blogging-prompt-card/prompts-navigation.jsx:
```diff
@@ -158,11 +158,22 @@ const PromptsNavigation = ( { siteId, prompts, tracksPrefix, index } ) => {
-							'"Post" here is a verb meaning "to publish", as in "post an answer to this writing prompt"',
-					} ) }
+				<Button
+					href={ getNewPostLink() }
```
In the case where the prompt is answered, should we link here to the editor to edit the post that is the answer rather than to ta new post?

Addison-Stavlo replied:
> should we link here to the editor to edit the post that is the answer

That could be good behavior if we know which post corresponds. Otherwise maybe just disabling the button is a good approach?

roo2 replied:
yea we don't seem to have the post link handy, I'm not sure how we'd get it. Ahh yep, I see "option C" was to have it link to their answer 👍. I'll just change it to "Option B" which was disabling the link

eoigal reviewed (commented):
This worked for me, just have one question about clicking the Answered button. I kinda expected this to load the post I answered this prompt to, not a new post.

roo2 committed:
disable button if prompt answered

crisbusquets commented:
> This worked for me, just have one question about clicking the Answered button. I kinda expected this to load the post I answered this prompt to, not a new post.

Yep, me too. That was the initial propose I shared. Is that possible, @roo2 ?

roo2 committed:
fix lint

roo2 commented:
Hey @crisbusquets, I couldn't get it to link to the existing answer because we don't have that data on the frontend.

Addison-Stavlo commented:
✅ Noting I communicated approval of the current state before being able to do a direct approval here.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9652268

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @roo2 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9652268).
